### PR TITLE
Login logout refactor tyler

### PR DESCRIPTION
**Pull request by tylerpporter**
- Creates `visitor?` helper method in ApplicationController 
- Renames SessionsController to singleton (SessionController)
- Creates multiple private helper methods in SessionController
- Refactors SessionController 
- Refactors routes using `resources`

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -11,15 +11,15 @@
...
       <%= link_to "Cart: #{cart.total_items}", "/cart" unless current_admin? %>
-      <%= link_to "Login", "/login" if current_user.nil? %> <!-- I need to not see this -->
-      <%= link_to "Register", "/register" if current_user.nil? %> <!-- I need to not see this -->
+      <%= link_to "Login", "/login" if visitor? %>
+      <%= link_to "Register", "/register" if visitor? %>
       <%= link_to "Home", "/" %>
-      <%= link_to "My Profile", "/profile" unless current_user.nil? %>
-      <%= link_to "Logout", "/logout", method: :delete unless current_user.nil? %>
+      <%= link_to "My Profile", "/profile" unless visitor? %>
+      <%= link_to "Logout", "/logout", method: :delete unless visitor? %>
```
Comment (cmpprg) [inline app/views/layouts/application.html.erb:22]: Thanks for taking care of this

**Review by cmpprg — APPROVED** [review]
Looks clean.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -11,15 +11,15 @@
   <body>
 
     <nav class = "topnav">
-      <%= "Logged in as #{current_user.name}" unless current_user.nil?  %>
+      <%= "Logged in as #{current_user.name}" unless visitor? %>
```
Comment (tony-gomes) [inline app/views/layouts/application.html.erb:14]: really clean.

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (tony-gomes) [reply app/views/layouts/application.html.erb:22]: ditto. really clean.

**Review by tony-gomes — APPROVED** [review]
I dig all these changes... the resources in the controller is super clean.

File: app/controllers/session_controller.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  def new
+    flash[:error] = "You are already logged in" if current_user.present?
+    user_redirect
+  end
+
+  def create
+    user = User.find_by(email: params[:email])
+    authenticated?(user) ? login(user) : invalid_user
+  end
```
Comment (Shock253) [inline app/controllers/session_controller.rb:11]: This makes the rest of this controller much more readable, great code

**Review by Shock253 — APPROVED** [review]

**Pull request merged**